### PR TITLE
Improve content in Alter workout SQL DDL

### PR DESCRIPTION
**Pull request by kapnobatai137**
1. Add Column:
- introduced syntax for adding a column that is also a `PRIMARY KEY`
- changed tables to md format

2. Add Constraint:
- introduced syntax for adding a `FOREIGN KEY` constraint
- changed tables to md format
- introduced syntax for dropping a constraint

3. Add Default Value:
- changed title to Alter Characteristics
- introduced alter data type concept
- added example syntax for alter data type
- introduced rules for altering data type
- changed add default value syntax to ANSI Standard
- introduced drop default value concept
- introduced syntax for adding not null clause
- introduced syntax for dropping not null clause
- changed tables to md format
- improved revision question to request adding a new default value and dropping a not null clause

4. Add trigger:
- changed syntax to ANSI Standard
- Added `ON UPDATE` and `ON DELETE` concepts
- added argument concepts
  - `CASCADE`
  - `SET NULL`
  - `SET DEFAULT`
  - `RESTRICT`
  - `NO ACTION`
- modified revision question to fit new syntax
- wrote new practice question
  - 5 correct/10 total answers

5. Add rule:
- insight was removed due to being too advanced and *flavour* specific (PostgreSQL and MySQL Server only)
- it will be added in a future workout that will contain differences between ANSI Standard and each *flavour*

### CODE REVIEW

**Comment by enki-hq** [comment]
Rule insight:code-width failed for files: 
  - sql/ddl/alter/add-column.md

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/.archived/sql/ddl/alter/add-default-value.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/.archived/sql/ddl/alter/add-default-value.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/.archived/sql/ddl/alter/add-rule.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/.archived/sql/ddl/alter/add-rule.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/sql/ddl/alter/add-column.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/sql/ddl/alter/add-column.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/sql/ddl/alter/add-constraint.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/sql/ddl/alter/add-constraint.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/sql/ddl/alter/add-trigger.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/sql/ddl/alter/add-trigger.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)
- [ ] [enkidevs/curriculum/sql/ddl/alter/alter-characteristics.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/sql/ddl/alter/alter-characteristics.md?ref=13c83236e8d1d392a6c3ec3d9392bbb5d9344a6f)

**Comment by kapnobatai137** [comment]
Add trigger:
- split into 2 insights

Add Trigger (1):
- create trigger syntax
- `RESTRICT` argument
- `NO ACTION` argument
- fixed practice question to fit new content
- removed revision question
  - no need to have 2 revision questions on this topic

Add Trigger (2):
- `CASCADE` argument
- `SET NULL` argument
- `SET DEFAULT` argument
- modified practice question to fit new content
- slightly improved revision question to contain content from `Add Trigger (1)`

File: sql/ddl/alter/add-trigger-2.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+# Add Trigger (2)
+
+---
+## Content
+
+In the previous insight we showed you two arguments, `RESTRICT` and `NO ACTION`, which are used to restrict deleting records that have a matching row in the child tables. It's also possible to create triggers that modify the data in a table using arguments such as `CASCADE`, `SET NULL`, and `SET DEFAULT`.
+
+### CASCADE
+
+Using the `CASCADE` argument means that when a row in the parent table is deleted or updated, all the matching rows in the child tables are deleted or updated. Here is an example syntax:
```
Comment (nem035) [inline sql/ddl/alter/add-trigger-2.md]: Maybe add a "too" at the end of the first sentence?

Using X says that when parent is updated, the child is updated too.

File: sql/ddl/alter/add-trigger-2.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+* ON DELETE
+* RESTRICT
+* NO ACTION
+* SET DEFAULT
+* SET NULL
+
+---
+## Revision
+
+Complete the following syntax such that the *trigger* named 'cstr_2' does the following:
```
Comment (nem035) [inline sql/ddl/alter/add-trigger-2.md]: Should we say `cstr_2` instead of 'cstr_2'?

File: sql/ddl/alter/add-trigger-2.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+- when deleting from the parent table the corresponding record in the child table is set to its default value
+- when updating the parent table the corresponding record in the child table is updated
+
+prohibits deleting a record from the `pokemons` table if there are any matching records in the `region` table.
```
Comment (nem035) [inline sql/ddl/alter/add-trigger-2.md]: Is this sentence missing a beginning phrase?

Comment (kapnobatai137) [reply sql/ddl/alter/add-trigger-2.md]: I somehow forgot to completely delete the previous question

**Pull request merged**